code/datums/components/moved_relay.dm

```diff
@@ -0,0 +1,60 @@
+/**
+ * This component will ensure that move calls from parent locs
+ * will be relayed to the parent atom.
+ * This means that something inside of another object can
+ * still know when that other object moves.
+ */
+/datum/component/moved_relay
+	//List of ordered parents
+	//Index 1: parent
+	//Index 2: Parent's parent
+	//etc.
+	var/list/atom/ordered_parents = list()
+
+/datum/component/moved_relay/Initialize(...)
+	var/atom/A = parent
+	//Start tracking from the parent
+	//We will relay the parents move to itself for convenience
+	//if the super parent gets deleted, everything dies
+	RegisterSignal(A, COMSIG_MOVABLE_MOVED, .proc/parent_moved)
+	ordered_parents += A
+	//Recursively register parents
+	if(A.loc && !isturf(A.loc))
+		register_parent(A.loc)
+
+/datum/component/moved_relay/Destroy(force, silent)
+	for(var/atom/A as() in ordered_parents)
```

Comment:
I don't think this is safe. If our parent is getting destroyed, he can have nulls in contents.
```suggestion
	for(var/atom/A in ordered_parents)
```

Reply:
This list is safe, as I created it. Everything inside this list has a signal registered for deletion, so is hard-del safe.
![image](https://user-images.githubusercontent.com/26465327/192119968-6e1c4820-2331-40da-bbf9-8b80f0bc7080.png)